client/client-lib/src/lib.rs

```diff
@@ -105,7 +106,7 @@ impl From<GatewayClientConfig> for LightningGateway {
         LightningGateway {
             mint_pub_key: config.redeem_key.public_key(),
             node_pub_key: config.node_pub_key,
-            api: config.api,
+            api: config.api.to_string(),
```

Comment:
sounds silly but maybe add a `//TODO: use Url type in LightningGateway`
(I know you will do this in later PRs but like this nothing can be forgotten xd)

Reply:
The type system is going to complain once `api`'s type change.

Reply:
I'm not sure if I understand @dpc

Reply:
I don't think there's a need to leave comments like this, as they are rather unnecessary. A github issue (or a TODO in an existing github issue) would probably be better/sufficient.